numpy/lib/function_base.py

```diff
@@ -1752,10 +1762,73 @@ def diff(a, n=1, axis=-1):
     slice1 = tuple(slice1)
     slice2 = tuple(slice2)
     if n > 1:
-        return diff(a[slice1]-a[slice2], n-1, axis=axis)
-    else:
+        return diff(a[slice1]-a[slice2], n-1, axis=axis, 
+            to_begin=to_begin, to_end=to_end)
+
+    # if nothing to add to either side, fast track the result
+    elif to_begin is None and to_end is None:
         return a[slice1]-a[slice2]
 
+    else:
+        # compute the length of the diff'd portion
+        # force length to be non negative
+        l_diff = a.shape[axis] - 1
+        if l_diff < 0:
+            l_diff = 0
+
+        # make to_end a 1D array
+        if to_end is None:
+            l_end = 0
+        else:
+            to_end = np.atleast_1d(to_end)
+            if to_end.ndim == 1:
+                l_end = len(to_end)
+            else:
+                to_end = np.array(to_end, ndmin=nd)
+                l_end = to_end.shape[axis]
+
+        # make to_begin a 1D array
+        if to_begin is None:
+            l_begin = 0
+        elif isinstance(to_begin, str) and to_begin == 'first':
```

Comment:
Is there a reason not to simply write `if to_begin == 'first':` (i.e., omit the `isinstance`). Python guarantees that equality checks never fail, i.e., that this evaluates to `False` if `to_begin` is not stringy.